samples/performance/wait_idle/wait_idle_tutorial.md

```diff
@@ -0,0 +1,75 @@
+
+The alternative to ``WaitIdle`` is to use a Vulkan ``Fence`` object, these are designed to allow the GPU to inform the CPU when it has finished with a single frame's workload allowing the CPU to safely re-use the resources for that frame. This method avoids stalling while waiting for the GPU to finish executing, as the CPU can continue to submit the following frames without having to wait for the GPU which in turn avoids the GPU pipeline draining of work.
+
+## The Wait Idle Sample
+
+This sample provides two radio buttons that allow you to alternate between using ``WaitIdle`` and ``Fence``.
+
+When ``WaitIdle`` is selected the sample calls ``vkDeviceWaitIdle`` before beginning each frame, this forces the GPU to finish executing all work dispatched to it and in doing so, drains the pipeline of all the work within. As a result, the GPU is idle while the next frame's command buffer is created until it has been dispatched, which increases frame times.
+
+When ``Fence`` is selected the sample assigns a ``Fence`` to each frame during its creation, then it calls ``vkWaitForFences`` and using the ``Fence`` for the next frame to be computed. This method allows the CPU to continue dispatching work to GPU while it executes the previous frames workload.
+
+Below is a screenshot of the sample running on a phone with a Mali G76 GPU:
+
+![Wait Idle Sample](images/wait_idle_sample.png)
+
+When ``WaitIdle`` is used the average frame time is 72ms, yet it decreases by 22% to 56ms when ``Fences`` is enabled.
+
+The performance increase is also clearly demonstrated when using [Streamline Performance Analyzer](https://developer.arm.com/products/software-development-tools/arm-development-studio/components/streamline-performance-analyzer).
+
+The graph below shows the Job Manager Cycles and stall in a single frame when ``WaitIdle`` is used.
+
+![Wait Idle Graph](images/wait_idle_graph.png)
+
+While, the following graph shows the Job Manager Cycles for a frame when ``Fences`` are used.
+
+![Fences Graph](images/fences_graph.png)
+
+This output clearly shows that ``WaitIdle`` forces the GPU to drain of all work which results in the GPU idling causing higher frame times.
+
+## Best practice summary
+
+**Do**
+
+* Use ``Fences`` to asynchronously read back data to the CPU; don't block synchronously and cause the pipeline to drain.
+
+**Don't**
+
+* Unnecessarily wait for GPU data on either the CPU or GPU.
+* Use ``vkQueueWaitIdle()`` or ``vkDeviceWaitIdle()``.
```

Comment:
Please elaborate on this. There are times where `vkQueueWaitIdle` and `vkDeviceWaitIdle` are perfectly acceptable blocking mechanism. This should explain that not use these when possible to use a less granular synchronization... This entire example needs to have more mentioning of this being used for the presentation engine as more compute orientated jobs have much different synchronization requirements